python/functional-programming/functional-programming-ii/README.md

```diff
@@ -0,0 +1,9 @@
+name: Functional Programming II
+
+description: Programming like mathematicians.
+
+insights:
+  - why-functional-programming-iv
+  - is-python-a-functional-language

```

Comment:
this insight has 390 words, so it can be split.

Functional Programing II should have these insights:
- is-python-a-functional-language-i
- is-python-a-functional-language-ii
- functional-particularities-of-python
- lambda-functions

Reply:
I've split and renamed it